src/class-parsely.php

```diff
@@ -380,6 +381,31 @@ public function get_options() {
 		);
 	}
 
+	/**
+	 * Sets the default values for the track_post_types and track_page_types
+	 * options.
+	 *
+	 * @since 3.9.0
+	 */
+	public function set_default_track_as_values(): void {
+		$this->option_defaults['track_page_types'] = array();
+		$this->option_defaults['track_post_types'] = array();
+
+		$post_types = get_post_types( array( 'public' => true ) );
+
+		foreach ( $post_types as $post_type ) {
+			if ( ! post_type_supports( $post_type, 'editor' ) ) {
+				continue;
+			}
+
+			if ( is_post_type_hierarchical( $post_type ) ) {
+				$this->option_defaults['track_page_types'][] = $post_type;
+			} else {
+				$this->option_defaults['track_post_types'][] = $post_type;
+			}
+		}
+	}
```

Comment:
Rather than changing the state of the object, I wonder if it's better to just calculate and return the configuration and then pass into the `array_merge`. This feels a bit cleaner but it's a bit subjective.

```suggestion
	public function get_default_post_type_tracking_config(): void {
		$defaults = array(
			'track_page_types' => array(),
			'track_post_types'  => array(),
		);

		$post_types = get_post_types( array( 'public' => true ) );

		foreach ( $post_types as $post_type ) {
			if ( ! post_type_supports( $post_type, 'editor' ) ) {
				continue;
			}

			if ( is_post_type_hierarchical( $post_type ) ) {
				$defaults['track_page_types'][] = $post_type;
			} else {
				$defaults['track_post_types'][] = $post_type;
			}
		}

		return $defaults;
	}
```

Reply:
I think I agree that this is cleaner.

However, this change has proven to be more complicated than I expected when I started working on it. We could implement it as part of a refactoring including some other things, but this is out of scope for this PR.